### PR TITLE
Neural Morphology for Translation Lookup

### PR DESCRIPTION
**Pull request by papajohn**

### CODE REVIEW

**Comment by papajohn** [comment]
See https://github.com/lilt/core/issues/412 for experimental results.

File: src/main/java/org/oscii/Main.java [hunk 1]
```diff
@@ -90,19 +94,28 @@ public static void main(String[] args) throws Exception {
...
+            if (options.has("embeddingsmorph")) {
+                morph = ((String) options.valueOf("embeddingsmorph")).split(",");
+            }
             if (langs.length != files.length) {
                 log.fatal("Unequal number of Word2Vec models ({}) and languages ({}).",
                           files.length, langs.length);
                 System.exit(-1);
             }
             embeddings = new Word2VecManager();
+            if (morph.length == langs.length) {
```
Comment (joewue) [inline src/main/java/org/oscii/Main.java]: there should be some feedback to the user that parameters don't match.

File: src/main/java/org/oscii/morph/MorphologyManager.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+     * @param language
+     * @return stem or original query if no stem is found
+     */
+    public String getKnownStem(String query, String language) {
+        Stemmer stemmer = stemmers.get(language);
+        List<String> stems = stemmer.getKnownStems(query);
+        if (stems.isEmpty()) {
+            return query;
+        } else {
+            return stems.get(0);
```
Comment (joewue) [inline src/main/java/org/oscii/morph/MorphologyManager.java:61]: Did you also try merging the results from multiple possible stems?

**Review by joewue — CHANGES_REQUESTED** [review]

File: src/main/java/org/oscii/morph/MorphologyManager.java (hunk 2 above)
Comment (papajohn) [reply src/main/java/org/oscii/morph/MorphologyManager.java:61]: Our interface only supports showing the translations of one term at the moment, so a merged set can't be displayed clearly.

File: src/main/java/org/oscii/Main.java (hunk 1 above)
Comment (papajohn) [reply src/main/java/org/oscii/Main.java]: Fixed

**Pull request merged**